frontend/app/src/connection/StaticConnection.tsx

```diff
@@ -0,0 +1,166 @@
+export async function getStaticConfig(): Promise<string> {
+  const isLocalStoreAvailable = localStorageAvailable()
+  let staticAssetUrl = ""
+
+  // Pull static asset url from localStorage if available
+  if (isLocalStoreAvailable) {
+    const cachedStaticAssetUrl =
+      window.localStorage.getItem("stStaticAssetUrl")
+    if (cachedStaticAssetUrl) {
+      return cachedStaticAssetUrl
+    }
+  }
+
+  // Otherwise, fetch url from config file
+  try {
+    const response = await fetch(STATIC_ASSET_CONFIG, {
+      signal: AbortSignal.timeout(5000),
+    })
+
+    if (!response.ok) {
+      logError("Failed to fetch static config url: ", response.status)
+    } else {
+      const config = await response.json()
+      staticAssetUrl = config.static_url ?? undefined
+
+      // Set in localStorage
+      if (isLocalStoreAvailable && staticAssetUrl) {
+        window.localStorage.setItem("stStaticAssetUrl", staticAssetUrl)
+      }
+    }
+  } catch (err) {
+    logError("Failed to fetch static config url:", err)
+  }
+
+  return staticAssetUrl
+}
+
+// Fetches FowardMsg protos from S3 for static streamlit apps
+// First, gets the location of the static assets from url in the config
+// Then fetches the protos from that location
+export async function getProtoResponse(
+  staticAppId: string
+): Promise<void | ArrayBuffer> {
+  const staticAssetURL = await getStaticConfig()
+
+  // Next, fetch the static app's protos (if we have a url)
+  if (staticAssetURL) {
+    const path = `${staticAssetURL}/${staticAppId}/protos.pb`
+    const response = await fetch(path, {
+      signal: AbortSignal.timeout(5000),
+    })
+
+    if (!response.ok) {
+      logError(
+        `Failed to fetch static app protos for id: ${staticAppId}`,
+        response.status
+      )
+    } else {
+      const data = await response.arrayBuffer()
+      return data
+    }
+  }
+
+  return
+}
+
+// Triggers fetch of static app assets and dispatches ForwardMsgs to be handled
+// by App.tsx's handleMessage, replicating the app
+export async function dispatchAppForwardMessages(
+  staticAppId: string,
+  onMessage: OnMessage,
+  onConnectionError: (message: string) => void
+): Promise<void> {
+  const arrayBuffer = await getProtoResponse(staticAppId)
+
+  if (!arrayBuffer) {
+    logError("Failed to retrieve static app protos")
+    onConnectionError(
+      `Failed to retrieve static app protos. Please confirm the id is correct and try again. Given static app id: ${staticAppId}`
+    )
+    return
+  }
+
+  // Transforms our arrayBuffer response into ForwardMsgList protos
+  const forwardMsgList = ForwardMsgList.decode(new Uint8Array(arrayBuffer))
+
+  // Dispatches each ForwardMsg to be handled by App.tsx's handleMessage
+  forwardMsgList.messages.forEach(msg => {
+    onMessage(msg)
+  })
+
+  return
+}
+
+export function establishStaticConnection({
+  staticAppId,
+  onConnectionStateChange,
+  onMessage,
+  onConnectionError,
+}: Props): void {
```

Comment:
This is fine, but since we are embracing the function, does this need to be an object of type Props? Why not just four parameters?